staging/src/k8s.io/client-go/tools/clientcmd/client_config.go

```diff
@@ -140,6 +141,16 @@ func (config *DirectClientConfig) ClientConfig() (*restclient.Config, error) {
 		clientConfig.Timeout = timeout
 	}
 
+	if configClusterInfo.ConnectionTimeout != "" {
+		connectionTimeout, err := ParseTimeout(configClusterInfo.ConnectionTimeout)
+		if err != nil {
+			return nil, err
+		}
+		clientConfig.ConnectionTimeout = connectionTimeout
+	} else {
+		clientConfig.ConnectionTimeout = 7 * time.Second
```

Comment:
Yes, adding a default changes the behavior of the entire client go library for all consumers and is not appropriate.

Also, I don't think a connection timeout is the first timeout to choose - we have a number of other timeouts already being set by other clients.